middlewares/ensure_guest_user.go

```diff
@@ -3,15 +3,25 @@ package middlewares
 import (
 	"net/http"
 
-	"github.com/gin-gonic/gin"
 	session "github.com/gutakk/go-google-scraper/helpers/session"
+	"github.com/gutakk/go-google-scraper/models"
+
+	"github.com/gin-gonic/gin"
 )
 
 func EnsureGuestUser(c *gin.Context) {
 	userID := session.Get(c, "user_id")
 
 	if userID != nil {
-		c.Redirect(http.StatusFound, "/")
+		_, err := models.FindUserByID(userID)
+
+		if err != nil {
+			session.Delete(c, "user_id")
+			c.Redirect(http.StatusFound, c.FullPath())
```

Comment:
What would be `c.FullPath()` 🤔 I am not sure if there is a need for different redirects

Reply:
Good catch. I did it in the complex way.
Fixed in 4e70d7f38f4c8664a2d3c60fb73f7474eebefd36